prototype2/common/HistSerializer.cpp

```diff
@@ -1,18 +1,22 @@
 /** Copyright (C) 2016, 2017 European Spallation Source ERIC */
 
-#include <cinttypes>
-#include <common/Trace.h>
 #include <common/HistSerializer.h>
+#include <libs/include/gccintel.h>
+
+#include <common/Trace.h>
 
 static_assert(FLATBUFFERS_LITTLEENDIAN,
               "Flatbuffers only tested on little endian systems");
 
-HistSerializer::HistSerializer()
-    : builder(2 * NMXHists::needed_buffer_size() + 256) {}
+HistSerializer::HistSerializer(size_t buffer_half_size)
+    : builder(2 * buffer_half_size + 256) {}
+
 
-HistSerializer::~HistSerializer() {}
+void HistSerializer::set_callback(ProducerCallback cb) {
```

Comment:
I think it is reasonable to require that you set the callback in the constructor.